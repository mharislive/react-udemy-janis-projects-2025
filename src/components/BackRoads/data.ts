import { faSocks, faTree, faWallet } from "@fortawesome/free-solid-svg-icons";
import tour1 from "../../assets/backroads/tour-1.jpeg";
import tour2 from "../../assets/backroads/tour-2.jpeg";
import tour3 from "../../assets/backroads/tour-3.jpeg";
import tour4 from "../../assets/backroads/tour-4.jpeg";

export const serviceData = [
  {
    title: "saving money",
    content:
      "Lorem ipsum dolor sit amet consectetur adipisicing elit. Asperiores, officia.",
    icon: faWallet,
  },
  {
    title: "endless hiking",
    content:
      "Lorem ipsum dolor sit amet consectetur adipisicing elit. Asperiores, officia.",
    icon: faTree,
  },
  {
    title: "amazing comfort",
    content:
      "Lorem ipsum dolor sit amet consectetur adipisicing elit. Asperiores, officia.",
    icon: faSocks,
  },
];

export const toursdata = [
  {
    content:
      "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Cumque vitae tempore voluptatum maxime reprehenderit eum quod exercitationem fugit, qui corporis.",
    image: tour1,
    date: "august 26th, 2020",
    title: "Tibet Adventure",
    country: "china",
    duration: "6 days",
    price: "$2100",
  },
  {
    content:
      "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Cumque vitae tempore voluptatum maxime reprehenderit eum quod exercitationem fugit, qui corporis.",
    image: tour2,
    date: "october 1th, 2020",
    title: "best of java",
    country: "indonesia",
    duration: "11 days",
    price: "$1400",
  },
  {
    content:
      "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Cumque vitae tempore voluptatum maxime reprehenderit eum quod exercitationem fugit, qui corporis.",
    image: tour3,
    date: "september 15th, 2020",
    title: "explore hong kong",
    country: "hong kong",
    duration: "8 days",
    price: "$5000",
  },
  {
    content:
      "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Cumque vitae tempore voluptatum maxime reprehenderit eum quod exercitationem fugit, qui corporis.",
    image: tour4,
    date: "december 5th, 2019",
    title: "kenya highlights",
    country: "kenya",
    duration: "20 days",
    price: "$3300",
  },
];
