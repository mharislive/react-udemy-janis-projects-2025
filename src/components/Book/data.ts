import type { BookType } from "../../types/Book";

export const BOOKS: BookType[] = [
  {
    title: "Atomic Habits",
    author: "James Clear",
    image: "https://m.media-amazon.com/images/I/81F90H7hnML._SY425_.jpg",
    id: 1,
  },
  {
    title: "Word Power Made Easy",
    author: "Norman Lewis",
    image: "https://m.media-amazon.com/images/I/71DortY2MXL._SY425_.jpg",
    id: 2,
  },
  {
    title: "The Psychology of Money Paperback",
    author: "Morgan Housel",
    image: "https://m.media-amazon.com/images/I/71XEsXS5RlL._SY425_.jpg",
    id: 3,
  },
];
