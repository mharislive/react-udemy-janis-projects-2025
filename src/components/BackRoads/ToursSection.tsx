import Container from "./Container";
import Section from "./Section";
import TourCard from "./TourCard";
import { toursdata } from "./data";

const ToursSection = () => {
  return (
    <Section title="Featured Tours" id="tours-section">
      <Container className="grid grid-cols-1 md:grid-cols-3 gap-8">
        {toursdata.map((item) => (
          <TourCard {...item} key={item.title} />
        ))}
      </Container>
    </Section>
  );
};

export default ToursSection;
