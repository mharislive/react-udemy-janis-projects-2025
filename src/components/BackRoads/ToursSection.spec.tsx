import { render, screen } from "@testing-library/react";
import { axe } from "vitest-axe";
import ToursSection from "./ToursSection";
import { toursdata } from "./data";

describe("BackRoads - ToursSection", () => {
  it("renders the section title", () => {
    render(<ToursSection />);
    expect(
      screen.getByRole("heading", { name: /Featured Tours/i, level: 2 })
    ).toBeInTheDocument();
  });

  it("renders a TourCard for each item in toursdata", () => {
    render(<ToursSection />);
    toursdata.forEach((tour) => {
      expect(screen.getByText(tour.title)).toBeInTheDocument();
    });
  });

  it("renders the correct number of TourCards", () => {
    render(<ToursSection />);
    const cards = toursdata.map((tour) => screen.getByText(tour.title));
    expect(cards).toHaveLength(toursdata.length);
  });

  it("should match snapshot", () => {
    const { asFragment } = render(<ToursSection />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should have no accessibility violations", async () => {
    const { container } = render(<ToursSection />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
