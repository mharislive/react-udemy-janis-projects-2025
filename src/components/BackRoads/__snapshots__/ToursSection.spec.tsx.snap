// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BackRoads - ToursSection > should match snapshot 1`] = `
<DocumentFragment>
  <section
    class="py-20 "
    id="tours-section"
  >
    <h2
      class="text-4xl font-bold text-center uppercase tracking-widest mb-16"
    >
      Featured 
      <span
        class="text-cyan-500"
      >
        Tours
      </span>
    </h2>
    <div
      class="container max-w-[1210px] mx-auto px-5 grid grid-cols-1 md:grid-cols-3 gap-8"
    >
      <div
        class="flex flex-col flex-1 shadow hover:shadow-2xl hover:scale-105 transition duration-500"
      >
        <div
          class="relative"
        >
          <img
            alt="Tibet Adventure"
            class="w-full"
            src="/src/assets/backroads/tour-1.jpeg"
          />
          <span
            class="absolute bottom-0 right-0 bg-cyan-300 px-2 py-1 capitalize"
          >
            august 26th, 2020
          </span>
        </div>
        <div
          class="py-5 px-6 bg-slate-100"
        >
          <h3
            class="font-bold tracking-widest mb-3 capitalize"
          >
            Tibet Adventure
          </h3>
          <p
            class="text-slate-500 mb-5"
          >
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Cumque vitae tempore voluptatum maxime reprehenderit eum quod exercitationem fugit, qui corporis.
          </p>
          <div
            class="flex items-center justify-between text-cyan-500"
          >
            <span
              class="capitalize"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-map "
                data-icon="map"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 576 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M384 476.1L192 421.2l0-385.3L384 90.8l0 385.3zm32-1.2l0-386.5L543.1 37.5c15.8-6.3 32.9 5.3 32.9 22.3l0 334.8c0 9.8-6 18.6-15.1 22.3L416 474.8zM15.1 95.1L160 37.2l0 386.5L32.9 474.5C17.1 480.8 0 469.2 0 452.2L0 117.4c0-9.8 6-18.6 15.1-22.3z"
                  fill="currentColor"
                />
              </svg>
               china
            </span>
            <span>
              6 days
            </span>
            <span>
              From $2100
            </span>
          </div>
        </div>
      </div>
      <div
        class="flex flex-col flex-1 shadow hover:shadow-2xl hover:scale-105 transition duration-500"
      >
        <div
          class="relative"
        >
          <img
            alt="best of java"
            class="w-full"
            src="/src/assets/backroads/tour-2.jpeg"
          />
          <span
            class="absolute bottom-0 right-0 bg-cyan-300 px-2 py-1 capitalize"
          >
            october 1th, 2020
          </span>
        </div>
        <div
          class="py-5 px-6 bg-slate-100"
        >
          <h3
            class="font-bold tracking-widest mb-3 capitalize"
          >
            best of java
          </h3>
          <p
            class="text-slate-500 mb-5"
          >
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Cumque vitae tempore voluptatum maxime reprehenderit eum quod exercitationem fugit, qui corporis.
          </p>
          <div
            class="flex items-center justify-between text-cyan-500"
          >
            <span
              class="capitalize"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-map "
                data-icon="map"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 576 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M384 476.1L192 421.2l0-385.3L384 90.8l0 385.3zm32-1.2l0-386.5L543.1 37.5c15.8-6.3 32.9 5.3 32.9 22.3l0 334.8c0 9.8-6 18.6-15.1 22.3L416 474.8zM15.1 95.1L160 37.2l0 386.5L32.9 474.5C17.1 480.8 0 469.2 0 452.2L0 117.4c0-9.8 6-18.6 15.1-22.3z"
                  fill="currentColor"
                />
              </svg>
               indonesia
            </span>
            <span>
              11 days
            </span>
            <span>
              From $1400
            </span>
          </div>
        </div>
      </div>
      <div
        class="flex flex-col flex-1 shadow hover:shadow-2xl hover:scale-105 transition duration-500"
      >
        <div
          class="relative"
        >
          <img
            alt="explore hong kong"
            class="w-full"
            src="/src/assets/backroads/tour-3.jpeg"
          />
          <span
            class="absolute bottom-0 right-0 bg-cyan-300 px-2 py-1 capitalize"
          >
            september 15th, 2020
          </span>
        </div>
        <div
          class="py-5 px-6 bg-slate-100"
        >
          <h3
            class="font-bold tracking-widest mb-3 capitalize"
          >
            explore hong kong
          </h3>
          <p
            class="text-slate-500 mb-5"
          >
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Cumque vitae tempore voluptatum maxime reprehenderit eum quod exercitationem fugit, qui corporis.
          </p>
          <div
            class="flex items-center justify-between text-cyan-500"
          >
            <span
              class="capitalize"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-map "
                data-icon="map"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 576 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M384 476.1L192 421.2l0-385.3L384 90.8l0 385.3zm32-1.2l0-386.5L543.1 37.5c15.8-6.3 32.9 5.3 32.9 22.3l0 334.8c0 9.8-6 18.6-15.1 22.3L416 474.8zM15.1 95.1L160 37.2l0 386.5L32.9 474.5C17.1 480.8 0 469.2 0 452.2L0 117.4c0-9.8 6-18.6 15.1-22.3z"
                  fill="currentColor"
                />
              </svg>
               hong kong
            </span>
            <span>
              8 days
            </span>
            <span>
              From $5000
            </span>
          </div>
        </div>
      </div>
      <div
        class="flex flex-col flex-1 shadow hover:shadow-2xl hover:scale-105 transition duration-500"
      >
        <div
          class="relative"
        >
          <img
            alt="kenya highlights"
            class="w-full"
            src="/src/assets/backroads/tour-4.jpeg"
          />
          <span
            class="absolute bottom-0 right-0 bg-cyan-300 px-2 py-1 capitalize"
          >
            december 5th, 2019
          </span>
        </div>
        <div
          class="py-5 px-6 bg-slate-100"
        >
          <h3
            class="font-bold tracking-widest mb-3 capitalize"
          >
            kenya highlights
          </h3>
          <p
            class="text-slate-500 mb-5"
          >
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Cumque vitae tempore voluptatum maxime reprehenderit eum quod exercitationem fugit, qui corporis.
          </p>
          <div
            class="flex items-center justify-between text-cyan-500"
          >
            <span
              class="capitalize"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-map "
                data-icon="map"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 576 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M384 476.1L192 421.2l0-385.3L384 90.8l0 385.3zm32-1.2l0-386.5L543.1 37.5c15.8-6.3 32.9 5.3 32.9 22.3l0 334.8c0 9.8-6 18.6-15.1 22.3L416 474.8zM15.1 95.1L160 37.2l0 386.5L32.9 474.5C17.1 480.8 0 469.2 0 452.2L0 117.4c0-9.8 6-18.6 15.1-22.3z"
                  fill="currentColor"
                />
              </svg>
               kenya
            </span>
            <span>
              20 days
            </span>
            <span>
              From $3300
            </span>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
